receiver/stanzareceiver/factory.go

```diff
@@ -17,66 +17,59 @@ package stanzareceiver
 import (
 	"context"
 
-	stanza "github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/pipeline"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver/receiverhelper"
 )
 
-const (
-	typeStr = "stanza"
-	verStr  = "0.12.0"
-)
-
-// NewFactory creates a factory for Stanza receiver
-func NewFactory() component.ReceiverFactory {
-	return receiverhelper.NewFactory(
-		typeStr,
-		createDefaultConfig,
-		receiverhelper.WithLogs(createLogsReceiver))
+// LogReceiverType is the interface used by stanza-based log receivers
+type LogReceiverType interface {
+	configmodels.Receiver
```

Comment:
Why are we embedding `configmodels.Receiver`? From what I can tell we never use `Name()` or `SetName()`, we only use `Type()`. If that's true just add `Type()` here?